packages/vite/src/node/ssr/ssrTransform.ts

```diff
@@ -31,11 +31,16 @@ export async function ssrTransform(
 ): Promise<TransformResult | null> {
   const s = new MagicString(code)
 
-  const ast = parser.parse(code, {
-    sourceType: 'module',
-    ecmaVersion: 'latest',
-    locations: true
-  }) as any
+  let ast
+  try {
+    ast = parser.parse(code, {
+      sourceType: 'module',
+      ecmaVersion: 'latest',
+      locations: true
+    }) as any
+  } catch (err) {
+    throw new Error('Parse failure: ' + err.message + '\n' + code)
```

Comment:
I'm a bit worried about a log with the full `code`. Maybe we could log by default the `url` and have the `code` behind a debugger?

Reply:
Yeah, it is a bit big, but I'm not sure anything else would help the user that much. The `url` would not be helpful because the code has already been transformed by plugins and so the line and column number don't mean anything if you don't see the actual code.

At this point the page is failing to load entirely, so I think a user will not be that annoyed by a screen full of debug info because it's a lifeline to them in figuring out the error and that error is a much bigger problem than slightly noisy logs.

Reply:
Actually though, maybe I can just parse that line out of the code. Let me take a look. It looks like acorn sticks some info on the error: https://github.com/acornjs/acorn/blob/7deba41118d6384a2c498c61176b3cf434f69590/acorn/src/location.js#L16

Reply:
Ok. Fixed now!